### DISCUSSION TITLE
Add virtual actors (On/Off or Open/Close) and hide "real" ones

### DISCUSSION THREAD

**Discussion by qroac** [arendst/Tasmota#15336] [Q&A] [answered]
I have a scenario for my garden watering I'd very much like to solve with Tasmota.

I use latching valves, so I have to give a short impulse with reveresed polarity to open and close the valve.
To achieve this, I use a 4-Relais board, a ADS1115 to connect the relais with I2C and use 2 relais to set the polarity on the 2 outgoing wires and a 3rd to give the short pulse.

What I'd like is to hide these Relais from the webui as I do rather see a problem than a use in manual control of them.
Instead I'd like to add a "virtual actor", that shows a button "On/Off" or "Open/Close" and triggers not a physical output but a rule in which I can set the physical outputs as needed.

Is it possible to hide physical outputs in the webUI?
Is it possible to add a virtual actor that triggers a rule?
And would it be possible to trigger this virtual actor by MQTT, too?

**Comment by sfromis** [comment, marked as answer]
In general, the intention is that the web UI is only meant for admin/test, not for "regular operation", meaning that Tasmota does not provide much when it comes to tweaking what/how it appears, including user interface additions or subtractions. Daily operation is expected to be controlled from some backend setup. There are some web UI amendment features with scripting, or with Berry on ESP32. One "redneck" way of adding web UI buttons is to config virtual relays on unused pins, and have rules react to those changing state.

Since you need a "short impulse", did you see the setup for [PulseTime](https://tasmota.github.io/docs/Commands/#pulsetime)? You can control the pulsing without rules.

One way of invoking rules is using the [Event](https://tasmota.github.io/docs/Commands/#event) command, which can be issued from all command sources, including MQTT.

**Reply by qroac** [reply]
> the intention is that the web UI is only meant for admin/test, not for "regular operation

Understood. I think that defininitly makes sense for most use cases ;)

Knowing that, I'll focus on implementing it with Events triggered by node-red and PulseTimers.
Even if it doesn't work with the web-ui, I now have a much clearer idea how to implement it with MQTT
Thank you
